aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -929,57 +929,62 @@ internal class RealAWSCognitoAuthPlugin(
     ) {
         val forceRefresh = options.forceRefresh
         authStateMachine.getCurrentState { authState ->
-            when (val authZState = authState.authZState) {
-                is AuthorizationState.Configured -> {
-                    authStateMachine.send(AuthorizationEvent(AuthorizationEvent.EventType.FetchUnAuthSession))
-                    _fetchAuthSession(onSuccess, onError)
-                }
-                is AuthorizationState.SessionEstablished -> {
-                    val credential = authZState.amplifyCredential
-                    if (!credential.isValid() || forceRefresh) {
-                        if (credential is AmplifyCredential.IdentityPoolFederated) {
-                            authStateMachine.send(
-                                AuthorizationEvent(
-                                    AuthorizationEvent.EventType.StartFederationToIdentityPool(
-                                        credential.federatedToken,
-                                        credential.identityId,
-                                        credential
+            if (authState.authNState is AuthenticationState.SignedOut) {
```

Comment:
When authN is in SignedOut state, authZ will be either Configured or SessionEstablished. Are you seeing any other authZ state? What is authZ state when this fix is applicable?

Reply:
@gpanshu This is the root of the issue though. After a user has been deleted, the state should end up back in a settled state, ex Configured. If its not in a settled state, InvalidStateException is the right response.

Reply:
Each public AUTH api call is queued and only executes 1 at a time. This means that at the start of each call, the state machine should be fully settled. If we are seeing an issue with a state existing as DeletingUser, the delete user call is not moving properly to a final settled state.

Reply:
Yupp I also realized that and was working on a fix that would update the authZ state. This has been fixed. Please review again.